samples/net/http_server/src/config.h

```diff
@@ -30,4 +30,13 @@
 #define ZEPHYR_PORT		8080
 #endif
 
+#define HTTP_AUTH_URL		"/auth"
+#define HTTP_AUTH_TYPE		"Basic"
+
+/* HTTP Basic Auth, see https://tools.ietf.org/html/rfc7617 */
+#define HTTP_AUTH_REALM		"Zephyr"
+#define HTTP_AUTH_USERNAME	"zephyr"
+#define HTTP_AUTH_PASSWORD	"0123456789"
+#define HTTP_AUTH_CREDENTIALS	"emVwaHlyOjAxMjM0NTY3ODk="
```

Comment:
I know this is a sample, but we do have a `base64` implementation in the tree, part of mbedtls. Is it possible to use that, even if building a non-TLS version of the server?